Soft-pausing followed by hard pause exporting breaks the persisted exporting state

Steps to reproduce:

```
$ docker run --rm -p 9600:9600 -e CAMUNDA_DATA_SECONDARYSTORAGE_TYPE=none -e CAMUNDA_MONITORING_METRICS_ENABLEEXPORTEREXECUTIONMETRICS=false camunda/zeebe:8.8.27
$ xh POST :9600/actuator/exporting/pause soft==true
$ docker exec -t 7b9d27e85c4f cat data/raft-partition/partitions/1/.exporterPaused
SOFT_PAUSED
$ xh POST :9600/actuator/exporting/pause
docker exec -t 7b9d27e85c4f cat data/raft-partition/partitions/1/.exporterPaused
PAUSEDAUSED
```

Symptoms:
```
[2026-06-24 10:​33:​52.053] [zb-actors-0] [{actor-name=Startup, actor-scheduler=Broker-0}] ERROR
        io.camunda.zeebe.broker.partitioning.PartitionManagerImpl - Failed to start partition 1, removing partition and shutting down already started steps
io.camunda.zeebe.scheduler.startup.StartupProcessException: Startup failed in the following steps: [Partition 1 - Zeebe Partition]. See suppressed exceptions for details.
        at io.camunda.zeebe.scheduler.startup.StartupProcess.aggregateExceptionsSynchronized(StartupProcess.java:333)
        at io.camunda.zeebe.scheduler.startup.StartupProcess.completeStartupFutureExceptionallySynchronized(StartupProcess.java:221)
        at io.camunda.zeebe.scheduler.startup.StartupProcess.lambda$proceedWithStartupSynchronized$3(StartupProcess.java:203)
        at io.camunda.zeebe.scheduler.future.FutureContinuationRunnable.run(FutureContinuationRunnable.java:33)
        at io.camunda.zeebe.scheduler.ActorJob.invoke(ActorJob.java:86)
        at io.camunda.zeebe.scheduler.ActorJob.execute(ActorJob.java:43)
        at io.camunda.zeebe.scheduler.ActorTask.execute(ActorTask.java:125)
        at io.camunda.zeebe.scheduler.ActorThread.executeCurrentTask(ActorThread.java:127)
        at io.camunda.zeebe.scheduler.ActorThread.doWork(ActorThread.java:104)
        at io.camunda.zeebe.scheduler.ActorThread.run(ActorThread.java:224)
        Suppressed: io.camunda.zeebe.scheduler.startup.StartupProcessStepException: Bootstrap step Partition 1 - Zeebe Partition failed
                at io.camunda.zeebe.scheduler.startup.StartupProcess.completeStartupFutureExceptionallySynchronized(StartupProcess.java:223)
                ... 8 more
        Caused by: java.lang.IllegalArgumentException: No enum constant io.camunda.zeebe.broker.exporter.stream.ExporterPhase.PAUSEDAUSED
                at java.base/java.lang.Enum.valueOf(Unknown Source)
                at io.camunda.zeebe.broker.exporter.stream.ExporterPhase.valueOf(ExporterPhase.java:12)
                at io.camunda.zeebe.broker.system.partitions.impl.PartitionProcessingState.initExportingState(PartitionProcessingState.java:146)
                at io.camunda.zeebe.broker.system.partitions.impl.PartitionProcessingState.<init>(PartitionProcessingState.java:30)
                at io.camunda.zeebe.broker.partitioning.startup.ZeebePartitionFactory.constructPartition(ZeebePartitionFactory.java:197)
                at io.camunda.zeebe.broker.partitioning.startup.steps.ZeebePartitionStep.startup(ZeebePartitionStep.java:37)
                at io.camunda.zeebe.broker.partitioning.startup.steps.ZeebePartitionStep.startup(ZeebePartitionStep.java:15)
                at io.camunda.zeebe.scheduler.startup.StartupProcess.proceedWithStartupSynchronized(StartupProcess.java:191)
                at io.camunda.zeebe.scheduler.startup.StartupProcess.lambda$proceedWithStartupSynchronized$3(StartupProcess.java:205)
                at io.camunda.zeebe.scheduler.future.FutureContinuationRunnable.run(FutureContinuationRunnable.java:28)
                ... 6 more
```

Found in https://jira.camunda.com/browse/SUPPORT-33360

--- a/zeebe/broker/src/main/java/io/camunda/zeebe/broker/system/partitions/impl/PartitionProcessingState.java
+++ b/zeebe/broker/src/main/java/io/camunda/zeebe/broker/system/partitions/impl/PartitionProcessingState.java
@@ -126,7 +126,8 @@ public class PartitionProcessingState {
         state.name(),
         StandardCharsets.UTF_8,
         StandardOpenOption.DSYNC,
-        StandardOpenOption.CREATE);
+        StandardOpenOption.CREATE,
+        StandardOpenOption.TRUNCATE_EXISTING);
   }
 
   private void initExportingState() {
@@ -138,7 +139,7 @@ public class PartitionProcessingState {
         final var state =
             Files.readString(
                 getPersistedPauseState(PERSISTED_EXPORTER_PAUSE_STATE_FILENAME).toPath());
-        if (state == null || state.isEmpty() || state.isBlank()) {
+        if (state.isBlank()) {
           // Backwards compatibility. If the file exists, it is paused.
           exporterPhase = ExporterPhase.PAUSED;
           return;

--- a/zeebe/broker/src/test/java/io/camunda/zeebe/broker/system/partitions/impl/PartitionProcessingStateTest.java
+++ b/zeebe/broker/src/test/java/io/camunda/zeebe/broker/system/partitions/impl/PartitionProcessingStateTest.java
@@ -142,4 +142,18 @@ class PartitionProcessingStateTest {
         .describedAs("Exporting must be paused.")
         .isEqualTo(ExporterPhase.PAUSED);
   }
+
+  @Test
+  void shouldSupportSoftThenHardPause() {
+    // given
+    final var partitionProcessingState = new PartitionProcessingState(MOCK_RAFT_PARTITION);
+    partitionProcessingState.softPauseExporting();
+
+    // when
+    partitionProcessingState.pauseExporting();
+
+    // then
+    final var newState = new PartitionProcessingState(MOCK_RAFT_PARTITION);
+    assertThat(newState.getExporterPhase()).isEqualTo(ExporterPhase.PAUSED);
+  }
 }

